File: go/hamming/hamming.go
package hamming

import (
	"errors"
	"unicode/utf8"
)

func Distance(a, b string) (int, error) {
	len1 := utf8.RuneCountInString(a)
	len2 := utf8.RuneCountInString(b)

	if len1 != len2 {
		return 0, errors.New("provided strings do not have same length")
	}

	count := 0

	for i := 0; i < len1; i++ {
		if a[i] != b[i] {
			count++
		}
	}

	return count, nil
}

File: lua/hamming/hamming.lua
local Hamming = {}

function Hamming.compute(a,b)
	if string.len(a) ~= string.len(b) then return -1 end

	local count = 0

	for i=1,string.len(a) do
		if string.sub(a, i, i) ~= string.sub(b, i, i) then
			count = count + 1
		end
	end

	return count
end

return Hamming
